### PR TITLE
append commit SHA to commit msg, only if not a part of it already.

### PR DESCRIPTION
**Pull request by abhishek-kdm**
__*problem*__: user provided commit message being modified (appending the commit SHA).
As the user, it is a bit annoying to have the commit SHA appear twice in the commit message.

__*possible solutions:*__
* do not append the commit SHA, if user has already included it the commit message.
* let the user decide whether the user's commit message can be modified. *(maybe via argument in `action.yml`)*.

### CODE REVIEW

**Comment by JamesIves** [comment]
I wasn't aware you could pass in the commit SHA into an action arg. As that's the case we should probably make it so the commit message totally overrides the default one if it's specified instead of having logic that excludes it only if it's included.

**Comment by abhishek-kdm** [comment]
Yeah, that's the better option, the appended SHA is kinda redundant.

File: src/git.ts [hunk 1]
```diff
@@ -100,11 +100,9 @@ export async function deploy(action: ActionInterface): Promise<void> {
...
     hasRequiredParameters(action)
 
-    const commitMessage = `${
-      !isNullOrUndefined(action.commitMessage)
-        ? action.commitMessage
-        : `Deploying to ${action.branch} from ${action.baseBranch}`
-    } ${process.env.GITHUB_SHA ? `@ ${process.env.GITHUB_SHA}` : ''} 🚀`
+    const commitMessage = !isNullOrUndefined(action.commitMessage)
+      ? (action.commitMessage as string)
+      : `Deploying to ${action.branch} from ${action.baseBranch} @ ${process.env.GITHUB_SHA} 🚀`
```
Comment (JamesIves) [inline src/git.ts:105]:
```suggestion
      : `Deploying to ${action.branch} from ${action.baseBranch} ${process.env.GITHUB_SHA ? `@ ${process.env.GITHUB_SHA}` : ''} 🚀`
```

Let's keep the check here!

Comment (abhishek-kdm) [reply src/git.ts:105]: Yeah sure.
It always will be available though, they specified that in the docs.

**Comment by JamesIves** [comment]
Released as part of `3.4.7`: https://github.com/JamesIves/github-pages-deploy-action/releases/tag/3.4.7

Thanks!

**Pull request merged**